cdc/changefeed.go

```diff
@@ -712,6 +724,33 @@ func (c *changeFeed) calcResolvedTs(ctx context.Context) error {
 		}
 	}
 
+	//sync-point on
+	if c.info.SyncPointEnabled {
+		c.syncpointMutex.Lock()
+		defer c.syncpointMutex.Unlock()
+		if c.status.ResolvedTs == c.status.CheckpointTs && !c.updateResolvedTs {
+			log.Info("sync point reached by ticker", zap.Uint64("ResolvedTs", c.status.ResolvedTs), zap.Uint64("CheckpointTs", c.status.CheckpointTs), zap.Bool("updateResolvedTs", c.updateResolvedTs), zap.Uint64("ddlResolvedTs", c.ddlResolvedTs), zap.Uint64("ddlTs", c.ddlTs))
+			c.updateResolvedTs = true
+			err := c.sinkSyncpoint(ctx)
+			if err != nil {
+				log.Error("syncpoint sink fail", zap.Uint64("ResolvedTs", c.status.ResolvedTs), zap.Uint64("CheckpointTs", c.status.CheckpointTs))
+			}
+		}
+
+		if c.status.ResolvedTs == 0 {
+			c.updateResolvedTs = true
+		}
+
+		if c.status.ResolvedTs == c.status.CheckpointTs && c.status.ResolvedTs == c.ddlTs {
```

Comment:
Oh, found a bug here.
When we set `c.ddlTs` to `c.ddlJobHistory[0].BinlogInfo.FinishedTS`, and after `c.status.ResolvedTs == c.status.CheckpointTs` is true, it only means all processors have reached DDL resolved ts, and the DDL can be executed to downstream. But we can't ensure the DDL has been executed in downstream.

Reply:
That means we must wait the DDL been executed in downstream and then record the syncpoint in downstream.